src/common/config.cpp

```diff
@@ -71,6 +71,20 @@ void Config::initialize(ConfigParser const& cp) {
     }
   }
 
+  // guess --tsv-fields if not set
```

Comment:
I agree, this is brittle. Alternatives I was thinking about:
- The number of (non-fake) encoders and decoders, but models are being contructed much later than the config and their construction is independent from the corpus.
- The number of (non-zeroed) dim-vocabs loaded from the model special:model.yml, so this will not work for a new training, and LMs are a special case.

I changed the code adding the second approach, but for a fresh training the number of streams is still guessed based on the model type name.

Any ideas how we could determine the number of streams in a cleaner way?

Reply:
Unfortunately not...